models/Strategy.py

```diff
@@ -74,8 +84,16 @@ def isSellSignal(self) -> bool:
             if indicator not in self._df_last:
                 raise AttributeError(f"'{indicator}' not in Pandas dataframe")
 
+        # if core indicators(EMA and MACD) are disabled, do not sell
+        if self.app.disableBuyEMA() and self.app.disableBuyMACD() :
```

Comment:
I'd check all four of the indicators used in the current strategy. So you are missing OBV and ERI here. Then only warn if there are 0/4 indicators being used.

Reply:
but sell signal is only based on EMA and MACD.
assuming that only macd and ema are disabled, then the bot would always sell at next granularity unit, am i wrong ?

Reply:
This is the current buy strategy:

```
# criteria for a buy signal 1
if bool(self._df_last['ema12gtema26co'].values[0]) is True \
        and (bool(self._df_last['macdgtsignal'].values[0]) is True or self.app.disableBuyMACD()) \
        and (bool(self._df_last['goldencross'].values[0]) is True or self.app.disableBullOnly()) \
        and (float(self._df_last['obv_pc'].values[0]) > -5 or self.app.disableBuyOBV()) \
        and (bool(self._df_last['eri_buy'].values[0]) is True or self.app.disableBuyElderRay()) \
        and self.state.last_action != 'BUY': # required for all strategies
```

5 indicators, all of which are on by default.
Golden Cross, EMA Cross, MACD Cross, ERI is in bull, OBV is up momentum.

I see in your code you've turned off the ema sell signal, but you could potentially have buy signals, but the bot would never sell (if, for example, you'd disabled all options except golden cross and the bot made a purchase).

Reply:
Just to say, I do think this is a good idea, and thank you for your work on it.
We just need to be careful with the implementation of it (since people are staking real money with it).

Reply:
thank you for your help.
so, just to be clear : it  is preferable to keep ema sell signal even if ema buy signal is disabled ?

Reply:
i tried this:
-assuming ema and macd are core indicators, bot will not buy if both are disabled (warning displayed and isBuySignal function will return False).
-sell strategy unchanged

Reply:
Is this ready to be merged or are you still working on it?